packages/desktop-client/src/components/mobile/MobileAmountInput.js

```diff
@@ -0,0 +1,457 @@
+import { PureComponent, createContext, forwardRef } from 'react';
+
+import mitt from 'mitt';
+
+import {
+  toRelaxedNumber,
+  amountToCurrency,
+  getNumberFormat,
+} from 'loot-core/src/shared/util';
+
+import { colors } from '../../style';
+import { Button, Text, View } from '../common';
+
+function getValue(state) {
+  const { value } = state;
+  return value;
+}
+
+const AmountAccessoryContext = createContext(mitt());
+
+class AmountInput extends PureComponent {
+  static getDerivedStateFromProps(props, state) {
+    return { editing: state.text !== '' || state.editing };
+  }
+
+  constructor(props) {
+    super(props);
+    // this.backgroundValue = new Animated.Value(0);
+    this.backgroundValue = 0;
+
+    this.id = Math.random().toString().slice(0, 5);
+    this.state = {
+      editing: false,
+      text: '',
+      // These are actually set from the props when the field is
+      // focused
+      value: 0,
+    };
+  }
+
+  componentDidMount() {
+    if (this.props.focused) {
+      this.focus();
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.removeListeners) {
+      this.removeListeners();
+    }
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (!prevProps.focused && this.props.focused) {
+      this.focus();
+    }
+
+    if (prevProps.value !== this.props.value) {
+      this.setState({
+        editing: false,
+        text: '',
+        ...this.getInitialValue(),
+      });
+    }
+  }
+
+  parseText() {
+    return toRelaxedNumber(
+      this.state.text.replace(/[,.]/, getNumberFormat().separator),
+    );
+  }
+
+  // animate() {
+  //   this.animation = Animated.sequence([
+  //     Animated.timing(this.backgroundValue, {
+  //       toValue: 1,
+  //       duration: 1200,
+  //       useNativeDriver: true,
+  //     }),
+  //     Animated.timing(this.backgroundValue, {
+  //       toValue: 0,
+  //       duration: 1200,
+  //       useNativeDriver: true,
+  //     }),
+  //   ]);
+
+  //   this.animation.start(({ finished }) => {
+  //     if (finished) {
+  //       this.animate();
+  //     }
+  //   });
+  // }
+
+  onKeyPress = e => {
+    if (e.nativeEvent.key === 'Backspace' && this.state.text === '') {
+      this.setState({ editing: true });
+    }
+  };
+
+  getInitialValue() {
+    return {
+      value: Math.abs(this.props.value),
+    };
+  }
+
+  focus() {
+    this.input.focus();
+
+    const initialState = this.getInitialValue();
+    this.setState(initialState);
+
+    this.addEventListeners();
+
+    this.props.context.emit('reset');
+  }
+
+  addEventListeners() {
+    if (this.removeListeners) {
+      this.removeListeners();
+    }
+
+    this.props.context.on('start-math', this.onStartMath);
+    // This will be called in `onBlur`
+    this.removeListeners = () => {
+      this.props.context.off('start-math', this.onStartMath);
+      this.removeListeners = null;
+    };
+  }
+
+  onStartMath = op => {
```

Comment:
It seems like math isn’t supported anymore (and anyway it was kind of limited in the old mobile version since multiplication/division weren’t supported). How would you feel about removing all the math-related code to simplify this component?

Reply:
Agreed, that makes sense. Math support removed via https://github.com/actualbudget/actual/pull/1340/commits/1e72984926cdb2931f17a0ffb1ce1eac17c89170.